shell.nix

```diff
@@ -1,22 +1,23 @@
 with (import <nixpkgs> {});
```

Comment:
@ghedamat have a careful look at this for me. I just copypasta'd the solution and added the `pkgs.rustfmt` and the macos `darwin` stuff, but I want to make sure this works on all platforms :)